src/libcore/iter/adapters/chain.rs

```diff
@@ -54,7 +54,6 @@ impl<A, B> Iterator for Chain<A, B> where
 {
     type Item = A::Item;
 
-    #[inline]
```

Comment:
With lto (thin or fat), these attributes shouldn't really make much of a difference. Is the performance regression still there when these are present ?

Reply:
These attributes were the cause of the perf regression

Reply:
How is that possible? The PR that introduced the regression did not add these attributes, so they must have already been there, right ? That is, without that PR, these attributes were not introducing that regression, so something else in that PR would have caused it.